gratipay/models/participant.py

```diff
@@ -416,11 +417,11 @@ def add_email(self, email, resend_threshold='3 minutes'):
             nonce = self.db.one("""
                 UPDATE emails
                    SET verification_start=%s
-                 WHERE participant=%s
+                 WHERE participant_id=%s
                    AND address=%s
                    AND verified IS NULL
              RETURNING nonce
-            """, (verification_start, self.username, email))
+            """, (verification_start, self.id, email))
             if not nonce:
                 return self.add_email(email)
 
```

Comment:
Hmm. The Integrity error is being hit, which means a record exists. If a record exists though, we shouldn't be hitting the recursive loop... 

Given that the loop exited right there, we should be able to inspect the DB and see what's happening.. Open psql and try? Let's make sure we have the same setup though... here's mine

```bash
$ psql gratipay-test

gratipay-test=# \d emails
                                        Table "public.emails"
       Column       |           Type           |                      Modifiers
--------------------+--------------------------+-----------------------------------------------------
 id                 | integer                  | not null default nextval('emails_id_seq'::regclass)
 address            | text                     | not null
 verified           | boolean                  |
 nonce              | text                     |
 verification_start | timestamp with time zone | not null default now()
 verification_end   | timestamp with time zone |
 participant_id     | bigint                   |
Indexes:
    "emails_pkey" PRIMARY KEY, btree (id)
    "emails_address_verified_key" UNIQUE CONSTRAINT, btree (address, verified)
    "emails_participant_id_address_key" UNIQUE CONSTRAINT, btree (participant_id, address)
Check constraints:
    "verified_cant_be_false" CHECK (verified IS NOT FALSE)
Foreign-key constraints:
    "emails_participant_id_fkey" FOREIGN KEY (participant_id) REFERENCES participants(id) ON UPDATE RESTRICT ON DELETE RESTRICT

gratipay-test=#
```